src/components/Resources/ResourceTable.tsx

```diff
@@ -1,24 +1,59 @@
 import { useEffect, useState } from 'react'
 import { QueryData } from '@supabase/supabase-js'
 import client from '@/database/client.tsx'
-import TagSection from '@/components/TagSection.tsx'
 import { FaHeart } from 'react-icons/fa'
 import { Tooltip } from 'flowbite-react'
+import MiniTag from '@/components/Resources/MiniTag.tsx'
 
-export default function ResourceTable() {
-    const resourcesQuery = client.from('resources').select()
+export default function ResourceTable({
+    search,
+    filterTags,
+}: {
+    search: string
+    filterTags: string[]
+}) {
+    const resourcesQuery = client
+        .from('resources')
+        .select(
+            `id, name, link, description, num_helped, tag_resource(id, tags(id,name))`
+        )
     type ResourcesType = QueryData<typeof resourcesQuery>
 
     const [data, setData] = useState<ResourcesType>([])
+    const [filteredData, setFilteredData] = useState<ResourcesType>([])
+
     useEffect(() => {
         const fetchData = async () => {
             const { data, error } = await resourcesQuery
             if (error) throw error
             const resources: ResourcesType = data
             setData(resources)
+            setFilteredData(resources)
         }
         fetchData()
     }, [])
+
+    useEffect(() => {
+        setFilteredData(
+            data.filter(
+                (item) =>
+                    (search.length > 0 &&
+                        item?.name
+                            ?.toLowerCase()
+                            .includes(search.toLowerCase())) ||
+                    (search.length > 0 &&
+                        item?.description
+                            ?.toLowerCase()
+                            .includes(search.toLowerCase())) ||
+                    item?.tag_resource?.some((i) => {
+                        return (
+                            filterTags.length == 0 ||
+                            filterTags.includes(i?.tags?.name || '')
```

Comment:
I was envisioning the tags would &&ed together. So if you selected Crisis and Informational, it would only return resources with both those tags. Open to discussion tho!